### PR TITLE
Add tips for contributing, update State article

### PR DESCRIPTION
**Pull request by netnef**
For the State article, I made these changes:

* Convert the layout to a table (lmk if you don't like it)
* Update the info to match the latest progress
* Add links/issues to some features for reference

Preview: https://netnef.github.io/wiki/state/

### CODE REVIEW

**Comment by sharpenedblade** [comment]
> There is work in progress support for custom graphics on the Touch Bar.

Its not really WIP anymore, it should just say that its supported

> Trackpad  🟡 Partially working

It should be marked as working, but leave the note there. Same with audio.

**Comment by AdityaGarg8** [comment]
Wi-Fi and Bluetooth drivers are upstreamed. Also, Bluetooth is partially working for BCM4377 devices. On other devices too, it's not very stable.

**Review by AdityaGarg8 — CHANGES_REQUESTED** [review]
See latest comment

**Comment by AdityaGarg8** [comment]
I added a commit in your pr so that there is no clash.

File: docs/state.md
Comment (Redecorating) [inline docs/state.md:11]: maybe note that this relies on apple-bce

File: docs/state.md
Comment (Redecorating) [inline docs/state.md:8]: patch to fix overscan on intel gpu is not upstream yet afaik

File: docs/state.md
Comment (Redecorating) [inline docs/state.md:15]: can you note that this sonoma issue is in the apple-bce module (because suspend does work if you don't use that module)

File: docs/state.md
Comment (Redecorating) [inline docs/state.md:18]: using igpu no longer breaks resume so you can remove that bit. maybe also mention that toggling the dgpu power doesn't work

File: docs/state.md
Comment (Redecorating) [inline docs/state.md:13]: bluetooth is upstream

**Review by Redecorating — COMMENTED** [review]
having a table here is much better

**Comment by netnef** [comment]
I updated the State article to address all of the comments here. I also added "Thunderbolt" to the features list.

Preview: https://netnef.github.io/wiki/state/

There's one last thing I wanted to ask about:

> Some models have deadzones on the edges of their trackpads where swipes along the trackpad that start in these deadzones will not be registered.

Is this still an issue? Is there a list of affected models anywhere?

**Comment by AdityaGarg8** [comment]
> I updated the State article to address all of the comments here. I also added "Thunderbolt" to the features list.
> 
> Preview: https://netnef.github.io/wiki/state/
> 
> There's one last thing I wanted to ask about:
> 
> > Some models have deadzones on the edges of their trackpads where swipes along the trackpad that start in these deadzones will not be registered.
> 
> Is this still an issue? Is there a list of affected models anywhere?

Tbh not enough feedback there by the users. Today only mbp 16,2 got fixed.

**Comment by netnef** [comment]
Ok, then we can leave the notice there I guess. But I would still leave the status as "working" because for most models it works as well as it possibly can (using libinput, at least). We should probably also add details about how to use the Synaptics driver (with a disclaimer that it's unsupported and requires using deprecated X11 instead of Wayland). smileBeda provided instructions in the Discord for this. We can maybe paste those into a gist or wiki article.

**Comment by AdityaGarg8** [comment]
The gist is already in the wiki, you can put working as well

**Comment by AdityaGarg8** [comment]
Also, fix the lint errors please

**Comment by netnef** [comment]
> The gist is already in the wiki, you can put working as well

Do you have a link to that so I can include it here?

**Comment by AdityaGarg8** [comment]
https://wiki.t2linux.org/#guides-and-similar

I don't think we need to put it here though. But status as working should be put.

**Comment by netnef** [comment]
The reason I think we should reference it here is because of the note we have added to it:

> Works, but isn't as great as on macOS

On the other wiki page you linked, it says that you can obtain this:

> Implement macOS-like Keyboard and trackpad experience

I don't think adding a casual mention to it would hurt. Something like this:

> Works, but isn't as great as on macOS (no force touch or palm rejection). Some models have deadzones on the edges of their trackpads where swipes along the trackpad that start in these deadzones will not be registered. Using Synaptics + X11 as per [this guide](https://wiki.t2linux.org/#guides-and-similar) could make it feel more like macOS, but comes at the cost of using a legacy display server (X11) instead of Wayland, potentially exposing other issues, like blurry display scaling.

Let me know what you think.

**Comment by AdityaGarg8** [comment]
I leave this to other ma

> The reason I think we should reference it here is because of the note we have added to it:
> 
> > Works, but isn't as great as on macOS
> 
> On the other wiki page you linked, it says that you can obtain this:
> 
> > Implement macOS-like Keyboard and trackpad experience
> 
> I don't think adding a casual mention to it would hurt. Something like this:
> 
> > Works, but isn't as great as on macOS (no force touch or palm rejection). Some models have deadzones on the edges of their trackpads where swipes along the trackpad that start in these deadzones will not be registered. Using Synaptics + X11 as per [this guide](https://wiki.t2linux.org/#guides-and-similar) could make it feel more like macOS, but comes at the cost of using a legacy display server (X11) instead of Wayland, potentially exposing other issues, like blurry display scaling.
> 
> Let me know what you think.

I leave this to other maintainers @Redecorating @sharpenedblade

**Comment by sharpenedblade** [comment]
> https://gist.github.com/smileBeda/f0452f0d7f1f6d8aa772603411f7876f

This config doesnt seem to be doing anything that cant be done with libinput, we should try to get it working on wayland before puting this on the wiki (once its on the wiki people are going to use it and we "have" to support them). [touchpad preasure](https://wayland.freedesktop.org/libinput/doc/latest/touchpad-pressure-debugging.html) and [palm rejection](https://wayland.freedesktop.org/libinput/doc/latest/palm-detection.html) is supported, we just need to set the quirks, the touchpad edge bug is fixed with [this](https://github.com/t2linux/linux-t2-patches/commit/af351299c1bea3212fa934bb3eecef0c9a9e52a0) commit.

We can merge this now and change it later, or wait for proper libinput support. @Redecorating

File: docs/state.md
Comment (AdityaGarg8) [inline docs/state.md:20]: Is this still an issue?

File: docs/contribute.md
Comment (AdityaGarg8) [inline docs/contribute.md:20]: I messed up a line here, could you please correct it

**Review by AdityaGarg8 — COMMENTED** [review]
Let's merge this after addressing all this?

File: docs/state.md
Comment (sharpenedblade) [reply docs/state.md:20]: yes

**Comment by netnef** [comment]
> > https://gist.github.com/smileBeda/f0452f0d7f1f6d8aa772603411f7876f
> 
> This config doesnt seem to be doing anything that cant be done with libinput, we should try to get it working on wayland before puting this on the wiki (once its on the wiki people are going to use it and we "have" to support them). [touchpad preasure](https://wayland.freedesktop.org/libinput/doc/latest/touchpad-pressure-debugging.html) and [palm rejection](https://wayland.freedesktop.org/libinput/doc/latest/palm-detection.html) is supported, we just need to set the quirks, the touchpad edge bug is fixed with [this](https://github.com/t2linux/linux-t2-patches/commit/af351299c1bea3212fa934bb3eecef0c9a9e52a0) commit.
> 
> We can merge this now and change it later, or wait for proper libinput support. @Redecorating

The config is already on the wiki: https://wiki.t2linux.org/#guides-and-similar

Perhaps we should clarify that it's unsupported by the t2linux maintainers there.

**Comment by netnef** [comment]
I updated two pages to address concerns from @sharpenedblade. Let me know what you think.

* https://netnef.github.io/wiki/#guides-and-similar
* https://netnef.github.io/wiki/state/

**Comment by sharpenedblade** [comment]
The deadzone problem is fixed a few days ago, we can merge once thats removed

**Comment by netnef** [comment]
I removed the deadzones remark. @AdityaGarg8 are we good to merge?

**Comment by AdityaGarg8** [comment]
Looks like we are

**Comment by netnef** [comment]
👍 Thank you all for the help. Glad we got these pages updated. Long overdue.

**Pull request merged**
